tutor/src/screens/student-dashboard/index.jsx

```diff
@@ -24,6 +26,12 @@ class StudentDashboardShell extends React.Component {
   }
 
   render() {
+    // keep rendering loading screen if the user needs to agree to terms
+    // this way the screen stays the same without a flash of other content
+    if (User.terms_signatures_needed) {
```

Comment:
OK my one concern is that with the ModalManager if a tour triggers before the terms for some reason (it shouldn't right now but might eventually...) then you'll get a tour of the loading screen..

Reply:
hm, not quite, but close.  Since this is the main student dash, it wouldn't render any tour wrapper, which would prevent any tours from playing.  

So what we'd get is the same, but maybe with the old flash of content before terms if the manager did anything else

Reply:
Ah you're right... the tour will just return isReady == false and exit immediately and the terms will render... looks good then.